numpy/core/include/numpy/ndarraytypes.h

```diff
@@ -912,6 +918,8 @@ typedef int (PyArray_FinalizeFunc)(PyArrayObject *, PyObject *);
 #define NPY_ARRAY_UPDATEIFCOPY    0x1000 /* Deprecated in 1.14 */
 #define NPY_ARRAY_WRITEBACKIFCOPY 0x2000
 
+#define NPY_ARRAY_ENSURENOCOPY 0x4000
```

Comment:
Modify to bit shift later.

Reply:
Should this have an underscore at least, especially since right now I don't think we actually have support for it everywhere in the C-API?